SimPEG/electromagnetics/static/utils/static_utils.py

```diff
@@ -258,291 +297,260 @@ def geometric_factor(survey_object, space_type="half space", **kwargs):
-
-    if scale == "log":
-        cbar = plt.colorbar(
-            ph,
-            format="$10^{%.2f}$",
-            fraction=0.06,
-            orientation="horizontal",
-            ax=ax,
-            **cbar_opts,
-        )
-    elif scale == "linear":
-        cbar = plt.colorbar(
-            ph,
-            format="%.2f",
-            fraction=0.06,
-            orientation="horizontal",
-            ax=ax,
-            **cbar_opts,
-        )
-
-    ticks = np.linspace(vmin, vmax, 3)
-    cbar.set_ticks(ticks)
-    cbar.ax.tick_params()
-
-    # Plot apparent resistivity
-    if data_locations:
-        ax.plot(midx, midz, "k.", ms=1, alpha=0.4)
+        x_label = "Conductivity (S/m)"
 
-    if sameratio:
-        ax.set_aspect("equal", adjustable="box")
+    if len(thicknesses) < len(values):
+        thicknesses = np.r_[thicknesses, thicknesses[-1]]
+    z_grid = np.r_[0., np.cumsum(thicknesses)]
+    resistivity = np.repeat(values, 2)
+    rho_min = 0.9*np.min(values)
+    rho_max = 1.1*np.max(values)
 
-    if y_values == "n-spacing":
-        ticks = ax.get_yticks() * 2  # pseudo-depth divides by 2
-        spacing = np.abs(midz).min()
-        ax.set_yticklabels(np.around(-ticks / spacing, 1))
-        ax.set_ylabel("n-spacing")
-    elif y_values == "pseudo-depth":
-        ax.set_ylabel("pseudo-depth")
-
-    if data_type.lower() in DATA_TYPES["apparent conductivity"]:
-        cbar.set_label("Apparent Conductivity (S/m)")
-
-    elif data_type.lower() in DATA_TYPES["apparent resistivity"]:
-        cbar.set_label("Apparent Resistivity ($\\Omega$m)")
-
-    elif data_type.lower() in DATA_TYPES["potential"]:
-        cbar.set_label("Voltage (V)")
-
-    elif data_type.lower() in DATA_TYPES["apparent chargeability"]:
-        cbar.set_label("Apparent Chargeability (V/V)")
-
-    elif data_type.lower() in ["misfit", "misfitmap"]:
-        cbar.set_label("Misfit (V)")
+    z = []
+    for i in range(0, len(thicknesses)):
+        z.append(np.r_[z_grid[i], z_grid[i + 1]])
+    z = np.hstack(z)
+    
+    if plot_elevation:
+        y_label = "Elevation (m)"
+        z = z0 - z
+        z_grid = z0 - z_grid
+        flip_axis = False
+    else:
+        y_label = "Depth (m)"
+        flip_axis = True
+        
+    if ax == None:
+        fig = plt.figure(figsize=(6, 6))
+        ax = fig.add_axes([0.15, 0.15, 0.75, 0.75])
+    
+    if show_layers:
+        for locz in z_grid:
+            plt.plot(
+                np.linspace(rho_min, rho_max, 100),
+                np.ones(100) * locz,
+                "k--",
+                lw=0.5,
+                label='_nolegend_'
+            )
 
+    ax.plot(resistivity, z, **kwargs)
+    ax.set_xscale(scale)
+    ax.set_xlim(rho_min, rho_max)
+    if flip_axis:
+        ax.set_ylim(z.max(), z.min())
+    else:
+        ax.set_ylim(z.min(), z.max())
+    ax.set_xlabel(x_label)
+    ax.set_ylabel(y_label)
+    
     return ax
 
 
-def plot_3d_pseudosection(
+def plot_2d_pseudosection(
```

Comment:
Just asking a general question, but I personally associate pseudo sections as a 2D thing, so is it necessary to specifically say `plot_2d_pseudosection`? I understand the reasoning behind including it for 3D though (`plot_3d_...`), and I think the change in name is probably unnecessary.

Reply:
The input arguments of the plot_2d_pseudoection and the original plot_pseudosection are so different that I'm inclined to keep it as is for now.

Reply:
I think we leave the plot_2d_pseudosection function for now for the above reason. Then after 0.15.0, we can deprecate it and rename it to plot_pseudosection.